mtg_draft_ai/brains.py

```diff
@@ -128,6 +129,130 @@ def _common_neighbors_weighted(self, card_pool, candidate, colors):
+
+    def pick(self, pack, cards_owned, draft_info):
+        synergy_ratings = super()._ratings(pack, cards_owned, draft_info)
+
+        raw_combined_ratings = self._raw_combined_ratings(synergy_ratings, cards_owned)
+        normalized_ratings = self._normalize_ratings(raw_combined_ratings)
+        composite_ratings = self._composite_ratings(normalized_ratings)
+        composite_ratings.sort(key=lambda cr: cr.rating, reverse=True)
+
+        # replace full card object with just card name for more readable output
+        printable_candidates = [str(_CombinedRating(tup[0].name, *tup[1:])) for tup in composite_ratings]
+        print('\nrankings:\n{}'.format('\n'.join(printable_candidates)))
+
+        return pack[0] if len(composite_ratings) == 0 else composite_ratings[0].card
+
+
+
+    @classmethod
+    def _raw_combined_ratings(cls, synergy_ratings, cards_owned):
+        """Generate power ratings (power_delta, total_power) for each card."""
+
+        # Sum up total power rating of current cards owned for each color pair
+        pool_total_power_by_color_pair = {}
+        for color_pair in COLOR_PAIRS:
+            on_color_cards = [c for c in cards_owned if synergy.castable(c, color_pair)]
+            total_power = sum([cls._power_rating(c) for c in on_color_cards])
+            pool_total_power_by_color_pair[color_pair] = total_power
+
+        raw_ratings = []
+        for r in synergy_ratings:
+            power_delta = cls._power_rating(r.card)
+            total_power = pool_total_power_by_color_pair[r.color_combo] + power_delta
+
+            raw_combined_rating = _CombinedRating(r.card, r.color_combo, rating=None,
+                                                  power_delta=power_delta, total_power=total_power,
+                                                  edges_delta=r.edges_delta, total_edges=r.total_edges,
+                                                  common_neighbors_weighted=r.common_neighbors_weighted)
+            raw_ratings.append(raw_combined_rating)
+        return raw_ratings
+
+    @staticmethod
+    def _power_rating(card):
+        """Assign numerical power value for each power tier."""
+
+        # These values are pretty arbitrary, but they feel like reasonable defaults
+        # in lieu of a data-driven tuning process or theoretical basis for assigning them.
+        # TODO: it's possible we should read these directly from the tags instead, so the cube owner has full control.
+        values_by_tier = {
+            1: 1,
+            2: 0.8,
+            3: 0.6,
+            4: 0.3,
+            None: 0
+        }
+        if card.power_tier not in values_by_tier:
+            raise ValueError('Undefined power tier: {}'.format(card.power_tier))
+        return values_by_tier[card.power_tier]
+
+    @staticmethod
+    def _composite_ratings(normalized_ratings):
+        # Average all of the different power measures into a composite power rating.
```

Comment:
I sort of wonder if averaging the synergy ratings between different synergy elements in an unweighted way will incorrectly bias the bots toward picking cards you aren't realistically going to play. Ex, if you have 3 types of cards in your deck, Tokens, graveyard, and artifacts, but you have very few of the latter two and a lot of the former, you might end up seeing a strong tokens card rated lower than a strong card split somewhere between graveyard and artifacts.

Reply:
Hmm, in that scenario some of the times it's going to be better to double down on tokens, and sometimes it'd be better to go for more flexibility. I think it's hard/impossible to make basic heuristics which work well in all nuanced cases. At the same time, those might be judgment calls to competent human drafters as well.